lib/memory.gi

```diff
@@ -228,8 +228,10 @@ InstallMethod( \^, "objects with memory", true,
   function(a,b)
     local slp;
     slp := a!.slp;
-    if a!.n = 0 then
+    if a!.n = 0 or b = 0 then
       return ObjWithMemory( slp, 0, a!.el );
```

Comment:
My understanding of `a!.el` is that it is the group element in question. If `b = 0` but `a!.n <> 0` then `a!.el` can be something different from the identity.